mysql/datadog_checks/mysql/mysql.py

```diff
@@ -140,6 +141,8 @@ def __init__(self, name, init_config, instances):
         # go through the agent internal metrics submission processing those tags
         self._non_internal_tags = copy.deepcopy(self.tags)
         self.set_resource_tags()
+        with self._connect() as db:
+            self._is_innodb_engine_enabled = self._is_innodb_engine_enabled(db)
```

Comment:
I suggest to put this under `def check` after db connection is established https://github.com/DataDog/integrations-core/blob/8bce95701485f27b6d886c8c0cd26a493294e2c0/mysql/datadog_checks/mysql/mysql.py#L281. 
We can also rename the function `_is_innodb_engine_enabled` to something like `_check_innodb_engine_enabled`. 
In the `__init__` constructor, we can just initialize a variable `self.innodb_engine_enabled = None` as None to indicate the check has not been done.

Reply:
thx, for renaming will do.

as for moving it to check - that was my initial idea too. But I didn't like that I had to add db as an argument to runtime_queries.  Although, now, I greped the code and no one is calling this from outside of the class. I thought if it's without _ then it's public and I cant change the signature.

The other option is to modify ```_check_innodb_engine_enabled```, I could've check there if self.conn is None or not, and if not open a connection there. Pls let me know which one is better or something else.

What  I don't like in my solution and kind of in existing one is the absence of clarity - how one can know that ```_check_innodb_engine_enabled``` has to be called and not _is_innodb_engine_enabled. Same for the runtime_queries property vs _runtime_queries.